### PR TITLE
Fix crash in container run after pulling an image

### LINKED ISSUE #488: Segmentation fault on docker run after pulling remote image
**Description**

**Steps to reproduce the issue:**
<details>
<summary> Original issue </summary>

1. Build a package using the docker/cli@e636a538 and moby/moby@184cea5ff
2. Install package on ubuntu xenial
3. Run `docker run -d --name server --net testbind -p 8080:8080/udp appropriate/nc sh -c 'while true; do echo hello | nc -w 1 -lu 8080; done'`

</details>

1. Build a package using the docker/cli@e636a538 and moby/moby@184cea5ff
2. Install package on ubuntu xenial
3. Run `docker run <any_remote_image> echo hello`

EDIT:
It appears this is occurring after pulling **any** images from a remote repository

**Describe the results you received:**
<details>
<summary> Original issue </summary>

```
root@ubuntu-xenial:/vagrant# docker run -d --name server --net testbind -p 8080:8080/udp appropriate/nc sh -c 'while true; do echo hello | nc -w 1 -lu 8080; done'
Unable to find image 'appropriate/nc:latest' locally
latest: Pulling from appropriate/nc
12b41071e6ce: Pull complete
d41ebf7a19c0: Pull complete
0b849cb5c85c: Pull complete
a3ed95caeb02: Pull complete
Digest: sha256:ff0f2b54fb1e1bfe277047fd660415310232d301e27bec9423aa741de961e6d4
Status: Downloaded newer image for appropriate/nc:latest
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa8b605]

goroutine 1 [running]:
github.com/docker/cli/cli/command/container.runContainer(0xc4201cf740, 0xc4203a8150, 0xc420365400, 0xc420403ec0, 0x0, 0x0)
	/go/src/github.com/docker/cli/cli/command/container/run.go:170 +0xb75
github.com/docker/cli/cli/command/container.runRun(0xc4201cf740, 0xc42007d4d0, 0xc4203a8150, 0xc420365400, 0x0, 0x0)
	/go/src/github.com/docker/cli/cli/command/container/run.go:117 +0x39b
github.com/docker/cli/cli/command/container.NewRunCommand.func1(0xc420399d40, 0xc4203749a0, 0x4, 0xb, 0x0, 0x0)
	/go/src/github.com/docker/cli/cli/command/container/run.go:49 +0x109
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).execute(0xc420399d40, 0xc4200101c0, 0xb, 0xb, 0xc420399d40, 0xc4200101c0)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:646 +0x44e
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420366480, 0xc420366900, 0xc420331f10, 0xc42000e4a8)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:742 +0x349
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420366480, 0xc420366480, 0x161a780)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:695 +0x2b
main.main()
	/go/src/github.com/docker/cli/cmd/docker/docker.go:168 +0xc3
```

</details>

```
root@ubuntu-xenial:/vagrant# docker run centos:7 echo hello
Unable to find image 'centos:7' locally
7: Pulling from library/centos
74f0853ba93b: Pull complete
Digest: sha256:26f74cefad82967f97f3eeeef88c1b6262f9b42bc96f2ad61d6f3fdf544759b8
Status: Downloaded newer image for centos:7
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa8b605]

goroutine 1 [running]:
github.com/docker/cli/cli/command/container.runContainer(0xc4201cf740, 0xc4203aa150, 0xc420365400, 0xc420403de0, 0x0, 0x0)
	/go/src/github.com/docker/cli/cli/command/container/run.go:170 +0xb75
github.com/docker/cli/cli/command/container.runRun(0xc4201cf740, 0xc42007d4d0, 0xc4203aa150, 0xc420365400, 0x0, 0x0)
	/go/src/github.com/docker/cli/cli/command/container/run.go:117 +0x39b
github.com/docker/cli/cli/command/container.NewRunCommand.func1(0xc420397d40, 0xc420405650, 0x3, 0x3, 0x0, 0x0)
	/go/src/github.com/docker/cli/cli/command/container/run.go:49 +0x109
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).execute(0xc420397d40, 0xc4200100c0, 0x3, 0x3, 0xc420397d40, 0xc4200100c0)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:646 +0x44e
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420366480, 0xc420366900, 0xc420331f10, 0xc42000e4a8)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:742 +0x349
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420366480, 0xc420366480, 0x161a780)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:695 +0x2b
main.main()
	/go/src/github.com/docker/cli/cmd/docker/docker.go:168 +0xc3
```
**Describe the results you expected:**
<details>
<summary> Original issue </summary>

```
root@ubuntu-xenial:/home/ubuntu# docker run -d --name server --net testbind -p 8080:8080/udp appropriate/nc sh -c 'while true; do echo hello | nc -w 1 -lu 8080; done'
Unable to find image 'appropriate/nc:latest' locally
latest: Pulling from appropriate/nc
12b41071e6ce: Pull complete
d41ebf7a19c0: Pull complete
0b849cb5c85c: Pull complete
a3ed95caeb02: Pull complete
Digest: sha256:ff0f2b54fb1e1bfe277047fd660415310232d301e27bec9423aa741de961e6d4
113e0258b1defc42de45dba364b664eb861ca2b9351aab677aa72279f429a94b
```

</details>

```
root@ubuntu-xenial:/vagrant# docker run centos:7 echo hello
hello
```

**Additional information you deem important (e.g. issue happens only occasionally):**
Issue seems to disappear after remote image has been pulled.

```
root@ubuntu-xenial:/vagrant# docker run centos:7 echo hello
Unable to find image 'centos:7' locally
7: Pulling from library/centos
74f0853ba93b: Pull complete
Digest: sha256:26f74cefad82967f97f3eeeef88c1b6262f9b42bc96f2ad61d6f3fdf544759b8
Status: Downloaded newer image for centos:7
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa8b605]

goroutine 1 [running]:
github.com/docker/cli/cli/command/container.runContainer(0xc4201cf740, 0xc4203aa150, 0xc420365400, 0xc420403de0, 0x0, 0x0)
	/go/src/github.com/docker/cli/cli/command/container/run.go:170 +0xb75
github.com/docker/cli/cli/command/container.runRun(0xc4201cf740, 0xc42007d4d0, 0xc4203aa150, 0xc420365400, 0x0, 0x0)
	/go/src/github.com/docker/cli/cli/command/container/run.go:117 +0x39b
github.com/docker/cli/cli/command/container.NewRunCommand.func1(0xc420397d40, 0xc420405650, 0x3, 0x3, 0x0, 0x0)
	/go/src/github.com/docker/cli/cli/command/container/run.go:49 +0x109
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).execute(0xc420397d40, 0xc4200100c0, 0x3, 0x3, 0xc420397d40, 0xc4200100c0)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:646 +0x44e
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420366480, 0xc420366900, 0xc420331f10, 0xc42000e4a8)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:742 +0x349
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420366480, 0xc420366480, 0x161a780)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:695 +0x2b
main.main()
	/go/src/github.com/docker/cli/cmd/docker/docker.go:168 +0xc3
root@ubuntu-xenial:/vagrant# docker run centos:7 echo hello
hello
```

<details>
<summary> Original issue </summary>

Issue is consistent and is included in the test:

https://github.com/moby/moby/blob/184cea5ff710abde25547749e5608b24a255ba09/integration-cli/docker_cli_network_unix_test.go#L1812-L1822

</details>

**Output of `docker version`:**
Don't mind the `17.06.0-dev` it's from the base [moby/moby/VERSION](https://github.com/moby/moby/blob/master/VERSION) file

```
root@ubuntu-xenial:/vagrant# docker version
Client:
 Version:      17.06.0-dev
 API version:  1.31
 Go version:   go1.8.3
 Git commit:   184cea5ff
 Built:        Thu Aug 31 17:53:14 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.0-dev
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   184cea5ff
 Built:        Thu Aug 31 17:52:09 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
root@ubuntu-xenial:/vagrant# docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 17.06.0-dev
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-87-generic
Operating System: Ubuntu 16.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 992.2MiB
Name: ubuntu-xenial
ID: VPBN:G2KS:AESQ:RVVR:3WCT:PQL3:GRN3:QBD7:INWL:TUYA:54ZS:2TVB
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
This is running in a Vagrant box, system information:
```
root@ubuntu-xenial:/vagrant# cat /etc/os-release
NAME="Ubuntu"
VERSION="16.04.3 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.3 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```

### PR DESCRIPTION
**Pull request by dnephin**
Fixes #488

cc @seemethere

### CODE REVIEW

File: cli/command/trust.go [hunk 1]
```diff
@@ -12,6 +12,10 @@ var (
 	untrusted bool
 )
 
+func init() {
+	untrusted = !getDefaultTrustState()
```
Comment (dnephin) [inline cli/command/trust.go]: This is necessary to test any code that reads `command.IsTrusted()`. The default is trusted, which requires mocking of a bunch more client calls.

**Comment by seemethere** [comment]
Kicked an integration build, to verify: https://jenkins.dockerproject.org/job/docker-integration-tests/109/

**Comment by tiborvass** [comment]
LGTM

**Comment by seemethere** [comment]
Can confirm this remedies the test failures we were seeing from yesterday with the exception of `DockerSuite.TestRenameInvalidName`:

```
❯ bash ~/scripts/check_failures.sh https://jenkins.dockerproject.org/job/docker-integration-tests/109/consoleText
FOR: https://jenkins.dockerproject.org/job/docker-integration-tests/109/consoleText

Passed (thought would fail):
DockerSwarmSuite.TestServiceLogs
DockerSwarmSuite.TestSwarmNetworkPluginV2
DockerSwarmSuite.TestSwarmServicePsMultipleServiceIDs

Failed (should fail)
DockerSuite.TestRmiContainerImageNotFound
DockerSuite.TestRmiImageIDForceWithRunningContainersAndMultipleTags
DockerSuite.TestRunAttachDetachFromInvalidFlag
DockerSuite.TestRunWithNanoCPUs
DockerSuite.TestUpdateWithNanoCPUs

Failed (thought would pass):
DockerSuite.TestRenameInvalidName
```

<details>
<summary> DockerSuite.TestRenameInvalidName failure </summary>

```
00:33:06.330 ----------------------------------------------------------------------
00:33:06.330 FAIL: docker_cli_rename_test.go:72: DockerSuite.TestRenameInvalidName
00:33:06.330 
00:33:06.330 docker_cli_rename_test.go:81:
00:33:06.330     c.Assert(out, checker.Contains, "requires exactly 2 argument(s).", check.Commentf("%v", err))
00:33:06.330 ... obtained string = "" +
00:33:06.330 ...     "\"docker rename\" requires exactly 2 arguments.\n" +
00:33:06.330 ...     "See 'docker rename --help'.\n" +
00:33:06.330 ...     "\n" +
00:33:06.330 ...     "Usage:  docker rename CONTAINER NEW_NAME\n" +
00:33:06.330 ...     "\n" +
00:33:06.330 ...     "Rename a container\n"
00:33:06.330 ... substring string = "requires exactly 2 argument(s)."
00:33:06.330 ... 
00:33:06.330 Command:  /usr/local/cli/docker rename myname
00:33:06.330 ExitCode: 1
00:33:06.330 Error:    exit status 1
00:33:06.330 Stdout:   
00:33:06.330 Stderr:   "docker rename" requires exactly 2 arguments.
00:33:06.330 See 'docker rename --help'.
00:33:06.330 
00:33:06.330 Usage:  docker rename CONTAINER NEW_NAME
00:33:06.330 
00:33:06.330 Rename a container
00:33:06.330 
00:33:06.330 
00:33:06.330 Failures:
00:33:06.330 ExitCode was 1 expected 0
00:33:06.330 Expected no error
00:33:06.330 
00:33:06.892 
00:33:06.892 ----------------------------------------------------------------------
```

</details>

**Comment by dnephin** [comment]
Thanks, I'll open a PR in moby/moby to fix that test.

**Pull request merged**